roles/iommu/tasks/main.yml

```diff
@@ -3,11 +3,11 @@
   ansible.builtin.include_role:
     name: stackhpc.linux.grubcmdline
   vars:
-    kernel_cmdline: # noqa: var-naming[no-role-prefix]
-      - intel_iommu=on
+    kernel_cmdline: "{{ ['intel_iommu=on'] + (['vfio-pci.ids=' + vfio_pci_ids] if vfio_pci_ids is defined else []) }}"  # noqa: var-naming[no-role-prefix]
     kernel_cmdline_remove: # noqa: var-naming[no-role-prefix]
       - ^intel_iommu=
-  when: "'Intel' in ansible_facts.processor.0"
+      - ^vfio-pci\.ids=
+  when: ansible_facts.processor | select('search', 'Intel') | list | length > 0
```

Comment:
I believe `vfio-pci` is relevant for non-Intel processors too.

Reply:
Also I am not sure how the previous `when` clause even worked. Even on CentOS Stream 8, I see `ansible_facts.processor` looks like this:

```
  ansible_facts.processor:
  - '0'
  - GenuineIntel
  - Intel(R) Xeon(R) Gold 6146 CPU @ 3.20GHz
  - '1'
  - GenuineIntel
  - Intel(R) Xeon(R) Gold 6146 CPU @ 3.20GHz
```

So `ansible_facts.processor.0` is always a string set to `0`.

Reply:
I tested with different Ansible versions for Yoga and for 2023.1.

Reply:
cpu_facts come from /proc/cpuinfo - I've seen it returning 0 in the first line and GenuineIntel in the first line, maybe there was some bug in the past that caused lack of '0' in the first line